docs/slurm_usage.md

```diff
@@ -0,0 +1,169 @@
+# PHGv2 - SLURM Usage Guidelines for `align-assemblies`
+
+When running `align-assemblies` on a single machine, 
+the `--total-threads` and `--in-parallel` parameters are used to 
+determine how many threads to allocate for the alignment step and 
+how many genomes to run in parallel. Users only need to call the 
+`align-assemblies` command once and there is no pre-processing steps 
+necessary.
+
+However, for [HPC](https://en.wikipedia.org/wiki/High-performance_computing)
+systems leveraging the [SLURM](https://en.wikipedia.org/wiki/Slurm_Workload_Manager)
+job scheduler, parallel processes operate differently which will
+require slight modifications to the alignment step. In this document, 
+we will discuss how to use the `align-assemblies` command for usage 
+with the SLURM work scheduler system.
+
+## Workflow Overview
+
+### 1 - Reference preparation
+When running jobs in a SLURM array, the first step is to run the 
+`align-assemblies` command with the `--just-ref-prep` parameter set 
+to `true`. This will run the preliminary steps of aligning the 
+reference genome to the GFF CDS, creating the `reference-sam` and 
+`reference-cds-fasta` files needed for aligning the individual 
+assemblies. The output of this step will be written to the user 
+supplied output directory.
+
+### 2 - SLURM data array file creation
+The second step would be to create the SLURM data array file. This is 
+a text file which has an `align-assemblies` command, one per line, 
+for each assembly that will be run. It will use the `reference-sam` 
+and `reference-cds-fasta` files created in the first step as input to 
+the `align-assemblies` command. Users can create the file manually if 
+they prefer, but for long lists of assemblies, manual curation can be 
+tedious and prone to errors. PHGv2 provides a convenience command
+(`prepare-slurm-align-file`) to automatically generate these files. See 
+the ["Prepare SLURM File for Alignments"](#prepare-slurm-file-for-alignments)
+section for further details.
+
+### 3 - Job submission
+The third step is to submit the SLURM job. When running a SLURM job, 
+the `--in-parallel` parameter is not used as each `align-assemblies` 
+command is assigned to separate computer nodes in the HPC system. 
+Conversely, the number of threads to use is still specified by the 
+`--total-threads` parameter.
+
+
+## Prepare SLURM File for Alignments
+The `prepare-slurm-align-file` command is a convenience method for 
```

Comment:
Should we provide an example of both the file that this makes and a simple SLURM script that can be run to correctly do the Array?

Reply:
Added example script courtesy of @lynnjo.